frontend/src/metabase/dashboard/components/parameters/widgets/DateAllOptionsWidget.jsx

```diff
@@ -0,0 +1,149 @@
+    {testRegex: /^past([0-9]+)([a-z]+)s$/, deserialize: (matches) => {
+        return ["time-interval", noopRef, -parseInt(matches[0]), matches[1]]
+    }},
+    {testRegex: /^next([0-9]+)([a-z]+)s$/, deserialize: (matches) => {
+        return ["time-interval", noopRef, parseInt(matches[0]), matches[1]]
+    }},
+    {testRegex: /^this([a-z]+)$/, deserialize: (matches) => ["time-interval", noopRef, "current", matches[0]] },
+    {testRegex: /^~([0-9-T:]+)$/, deserialize: (matches) => ["<", noopRef, matches[0]]},
+    {testRegex: /^([0-9-T:]+)~$/, deserialize: (matches) => [">", noopRef, matches[0]]},
+    {testRegex: /^([0-9-T:]+)$/, deserialize: (matches) => ["=", noopRef, matches[0]]},
+    // TODO 3/27/17 Atte Keinänen
+    // Unify BETWEEN -> between, IS_NULL -> is-null, NOT_NULL -> not-null throughout the codebase
+    // $FlowFixMe
+    {testRegex: /^([0-9-T:]+)~([0-9-T:]+)$/, deserialize: (matches) => ["BETWEEN", noopRef, matches[0], matches[1]]},
+];
+
+function urlEncodedToFilter(urlEncoded: UrlEncoded): ?FieldFilter {
+    const deserializer =
+        deserializersWithTestRegex.find((des) => urlEncoded.search(des.testRegex) !== -1);
+
+    if (deserializer) {
+        const substringMatches = deserializer.testRegex.exec(urlEncoded).splice(1);
+        return deserializer.deserialize(substringMatches);
+    } else {
+        return null;
+    }
+}
+
+const prefixedOperators: [OperatorName] = ["Before", "After", "On", "Is Empty", "Not Empty"];
+function getFilterTitle(filter) {
+    const desc = generateTimeFilterValuesDescriptions(filter).join(" - ")
+    const op = getFilterOperator(filter);
+    const prefix = op && prefixedOperators.indexOf(op.name) !== -1 ? `${op.name} ` : "";
+    return prefix + desc;
+}
+
+type DefaultProps = {};
+type Props = {
+    setValue: (value: ?string) => void,
+    onClose: () => void
+};
+
+type State = { filter: FieldFilter };
+
+
+export default class DateAllOptionsWidget extends Component<DefaultProps, Props, State> {
+    state: State;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            // $FlowFixMe
+            filter: props.value != null ? urlEncodedToFilter(props.value) || [] : []
+        }
+    }
+
+    static propTypes = {};
+    static defaultProps = {};
+
+    static format = (urlEncoded: ?string) => {
+        if (urlEncoded == null) return null;
+        const filter = urlEncodedToFilter(urlEncoded);
+
+        return filter ? getFilterTitle(filter) : null;
+    };
+
+    commitAndClose = () => {
+        this.props.setValue(filterToUrlEncoded(this.state.filter));
+        this.props.onClose()
+    }
+
+    setFilter = (filter: FieldFilter) => {
+        this.setState({filter});
+    }
+
+    isValid() {
+        const filterValues = this.state.filter.slice(2);
+        return filterValues.every((value) => value != null);
+    }
+
+    render() {
+        return (<div>
```

Comment:
Might want to put a `min-width` here (or maybe on the popover in ParameterValueWidget?) to keep the size from jumping around so much. Check with @mazameli